nvdaHelper/espeak/speechPlayer.cpp

```diff
@@ -0,0 +1,137 @@
+#include <espeak-ng/espeak_ng.h>
+#include <espeak-ng/speak_lib.h>
+#include "speech.h"
+#include "klatt.h"
+#include "phoneme.h"
+#include "synthesize.h"
+#include "voice.h"
+#include <speechPlayer/src/speechPlayer.h>
+
+extern "C" WGEN_DATA wdata;
+extern "C" voice_t *wvoice;
+extern unsigned char *out_ptr;
+extern unsigned char *out_end;
+
+speechPlayer_handle_t speechPlayerHandle=NULL;
+#define minFadeLength 110
+
+inline bool needsMixWaveFile() {
+	return wdata.n_mix_wavefile>0;
+}
+
+unsigned int mixWaveFile(unsigned int maxNumSamples, sample* sampleBuf) {
+	unsigned int i=0;
+	for(;wdata.mix_wavefile_ix<wdata.n_mix_wavefile;++wdata.mix_wavefile_ix) {
+		if(i>=maxNumSamples) break;
+		int val;
+		if(wdata.mix_wave_scale==0) {
+			val=wdata.mix_wavefile[wdata.mix_wavefile_ix+wdata.mix_wavefile_offset];
+			++(wdata.mix_wavefile_ix);
+			signed char c=wdata.mix_wavefile[wdata.mix_wavefile_ix+wdata.mix_wavefile_offset];
+			val+=(c*256);
+		} else {
+			val=(signed char)wdata.mix_wavefile[wdata.mix_wavefile_ix+wdata.mix_wavefile_offset]*wdata.mix_wave_scale;

```

Comment:
please explicitly cast with `static_cast<signed char>(blah)` you can capture `blah` in whatever type it starts in with: `auto blah = wdata.mix.....`